### PR TITLE
Hb/ast

### PR DESCRIPTION
**Pull request by Howon**

### CODE REVIEW

**Review by aethos — APPROVED** [review]
👍

File: src/ast.mli [hunk 1]
```diff
@@ -1,7 +1,83 @@
...
-    | Lit of int
-    | Binop of expr * operator * expr
-    | Seq of expr * expr
-    | Asn of int * expr
-    | Var of int
+    | Binop of expr * bin_op * expr
+    | Bitop of int * bit_op * int
+    | Uop of u_op * expr
+    | Id of string
+    | Assign of expr * expr
```
Comment (vlad-scherbich) [inline src/ast.mli:62]: `Assign of string * expr` ?

File: src/ast.mli [hunk 2]
```diff
@@ -1,7 +1,83 @@
-type operator = Add | Sub | Mul | Div
+type bin_op =
+    | Add    | Minus | Multiply | Divide
```
Comment (vlad-scherbich) [inline src/ast.mli:2]: I don't think you need the first `|`. Same for many declarations below:
- `type types`
- `type bit_op`
- `type actor_op`
- `type expr`
- `type stmt`

File: src/ast.mli [hunk 3]
```diff
@@ -1,7 +1,83 @@
...
+
+type actor_type = Actor_t of string
+
+type pool_type = Pool_t of actor_type list
+
+type types =
+    | Primitive of p_type
+    | Container of cont_type
+    | Optional of opt_type
+    | Actor_t of actor_op
```
Comment (vlad-scherbich) [inline src/ast.mli:40]: Is `Actor_t` a special convention? It does not conform to the other names here, and I'd rather we be uniform, i.e. change this to `Actor`.

**Pull request closed without merging**